tests/test_core.py

```diff
@@ -63,15 +63,18 @@ def test_loop(self, env, mind, callback):
         callback.on_episode_end.assert_called_once_with(0, train_mode)
         callback.on_loop_end.assert_called_once_with(False)
 
-    class invalid_callback(Callback):
-        def on_step_taken(self, step, transition, info):
-            a = 1/0
+    @pytest.fixture
+    def invalid_callback(self):
+        mock = Mock(spec=Callback)
+        mock.on_step_taken.side_efect = BaseException
+
+        return mock
 
-    def test_loop_with_exception(self, env, mind, callback):
+    def test_loop_with_exception_during_execution(self, env, mind, invalid_callback):
         train_mode = False
         handled_exception = False
         try:
-            loop(env, mind, train_mode=train_mode, verbose=0, callbacks=[self.invalid_callback()])
-        except Exception:
+            loop(env, mind, train_mode=train_mode, verbose=0, callbacks=[invalid_callback])
+        except BaseException:
             handled_exception = True
         assert handled_exception
```

Comment:
sprawdź tu jeszcze odrazu, czy invalid_callback był wołany z on_loop_end i tam dostał true. To powinno jakoś tak wygladać:
```Python
[...]

assert invalid_callback.on_loop_end.assert_called_once_with(True)
assert handled_exception
```

Reply:
Done